test/testdata/rewriter/rbs_signatures_sigs.rb

```diff
@@ -0,0 +1,225 @@
+# typed: strict
+# enable-experimental-rbs-signatures: true
+
+extend T::Sig
+
+module P1; end
+module P2; end
+module P3; end
+module P4; end
+module P5; end
+module P6; end
+
+# Parse errors
+
+#: (P1, P2 -> void
+#          ^^ error: Failed to parse RBS signature (unexpected token for function parameter name)
+def parse_error1; T.unsafe(nil); end # error: The method `parse_error1` does not have a `sig`
+
+class ParseError2
+  #: void
+  #  ^^^^ error: Failed to parse RBS signature (expected a token `pARROW`)
+  def parse_error2(p1, p2); end # error: The method `parse_error2` does not have a `sig`
+
+  class ParseError3
+    # Some comment
+    #: v
+    #  ^ error: Failed to parse RBS signature (expected a token `pARROW`)
+    # Some comment
+    def parse_error3(p1, p2); end # error: The method `parse_error3` does not have a `sig`
+  end
+end
+
+  #:
+ #  ^ error: Failed to parse RBS signature (expected a token `pARROW`)
+  def parse_error4(p1, p2); end # error: The method `parse_error4` does not have a `sig`
+
+# Sigs
+
+# We do not create any RBI if there is no RBS sigs
+def method1; T.unsafe(nil); end # error: The method `method1` does not have a `sig`
+
+#: -> String
+def method2; T.unsafe(nil); end
+
+#: -> String
+def method3; T.unsafe(nil); end
+
+#:      -> String
+def method4; T.unsafe(nil); end
+
+# some comment
+#: -> String
+# some comment
+def method5; T.unsafe(nil); end
+
+  #: -> String
+# ^^^^^^^^^^^^ error: Unused type annotation. No method def before next annotation
+  #: -> void
+  def method6; T.unsafe(nil); end
+
+#: (P1) -> void
+def method7(p1)
+  T.reveal_type(p1) # error: Revealed type: `P1`
+end
+
+#: (P1, P2) -> void
+def method8(p1, p2)
+  T.reveal_type(p1) # error: Revealed type: `P1`
+  T.reveal_type(p2) # error: Revealed type: `P2`
+end
+
+#: (?Integer) -> void
+def method9(p1 = 42)
+  T.reveal_type(p1) # error: Revealed type: `Integer`
```

Comment:
There are rather a lot of tests with `T.reveal_type` in the methods that are supported to have RBS-rewritten sigs, but it looks like there are no test cases that actually call the RBS-rewritten sig'd methods.  Shouldn't we have some of the latter too?

Reply:
Good idea, I added a few tests showing the behaviours on calls 👍